contracts/story-nft/StoryNFTFactory.sol

```diff
@@ -0,0 +1,284 @@
+    }
+
+    // solhint-disable-next-line max-line-length
+    // keccak256(abi.encode(uint256(keccak256("story-protocol-periphery.StoryNFTFactory")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant StoryNFTFactoryStorageLocation =
+        0xf790322fec2c69d950299f25bd2b4e4f8b183652054d59cf2f75df434f22df00;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address ipAssetRegistry,
+        address licensingModule,
+        address piLicenseTemplate,
+        uint256 defaultLicenseTermsId,
+        address orgNft
+    ) {
+        if (
+            ipAssetRegistry == address(0) ||
+            licensingModule == address(0) ||
+            piLicenseTemplate == address(0) ||
+            orgNft == address(0)
+        ) revert StoryNFTFactory__ZeroAddressParam();
+
+        IP_ASSET_REGISTRY = ipAssetRegistry;
+        LICENSING_MODULE = licensingModule;
+        PILICENSE_TEMPLATE = piLicenseTemplate;
+        DEFAULT_LICENSE_TERMS_ID = defaultLicenseTermsId;
+        ORG_NFT = IOrgNFT(orgNft);
+
+        _disableInitializers();
+    }
+
+    /// @dev Initializes the contract.
+    /// @param accessManager The address of the protocol access manager.
+    /// @param signer The address of the StoryNFTFactory's whitelist signer.
+    /// @param rootOrgOwner The address of the owner of the root organization NFT and the root StoryNFT.
+    /// @param rootOrgName The name of the root organization.
+    /// @param rootOrgTokenURI The token URI of the root organization NFT.
+    /// @param rootStoryNftInitData The initialization data for the root StoryNFT (see {IStoryBadge-InitParams}).
+    function initialize(
+        address accessManager,
+        address signer,
+        address rootOrgOwner,
+        string calldata rootOrgName,
+        string calldata rootOrgTokenURI,
+        bytes calldata rootStoryNftInitData
+    ) external initializer {
+        if (accessManager == address(0)) revert StoryNFTFactory__ZeroAddressParam();
+        __AccessManaged_init(accessManager);
+        __UUPSUpgradeable_init();
+
+        // Mint the root organization NFT and register it as an IP
+        ORG_NFT.mintRootOrgNft(rootOrgOwner, rootOrgTokenURI);
+
+        StoryNFTFactoryStorage storage $ = _getStoryNFTFactoryStorage();
+        $.signer = signer;
+
+        // Deploys and initializes the root StoryNFT (StoryBadge)
+        StoryBadge rootStoryNft = new StoryBadge({
+            ipAssetRegistry: IP_ASSET_REGISTRY,
+            licensingModule: LICENSING_MODULE,
+            piLicenseTemplate: PILICENSE_TEMPLATE,
+            defaultLicenseTermsId: DEFAULT_LICENSE_TERMS_ID
+        });
+        rootStoryNft.initialize({
+            owner_: rootOrgOwner,
+            orgNft_: address(ORG_NFT),
+            orgTokenId_: 0,
+            orgIpId_: ORG_NFT.getRootOrgIpId(),
+            initData: rootStoryNftInitData
+        });
```

Comment:
We might not need to deploy StoryBadgeNFT for Story Org when initialize the factory.
I feel we should deploy an empty StoryBadegeNFT as default template separately and only need to pass in the address of default template in the contractor function.
Then the initilaize function only need accessManager and signer.
The BadgeNFT for Story can be deployed later with deployByAdmin.

Reply:
Resolved, and added deploy script! see: https://github.com/storyprotocol/protocol-periphery-v1/compare/725e774409b9137dabcc0776ea4e638640a03db8..61049137d2adb38b0949fbe7c258199d54d6df2e